storage/replica.go

```diff
@@ -368,6 +374,8 @@ func (r *Replica) Destroy(origDesc roachpb.RangeDescriptor) error {
 	}
 	// Clear the map.
 	r.mu.pendingCmds = map[storagebase.CmdIDKey]*pendingCmd{}
+	r.mu.internalRaftGroup = nil
+	r.mu.destroyed = roachpb.NewRangeNotFoundError(r.RangeID)
```

Comment:
`roachpb.NewError` returns a `*roachpb.Error` which does not implement the `error` interface. I used `util.Errorf` instead.